include/shuffle/BijectiveFunctionScanShuffle.h

```diff
@@ -3,149 +3,209 @@
 #include "ThrustInclude.h"
 #include "shuffle/Shuffle.h"
 #include <cuda.h>
+#include <moderngpu/kernel_compact.hxx>
+#include <moderngpu/kernel_scan.hxx>
 #include <numeric>
 
-namespace BijectiveScanFuncs
-{
-template <bool device>
-struct cached_allocator;
+namespace BijectiveScanFuncs {
+template <bool device> struct cached_allocator;
 
-template <>
-struct cached_allocator<true>
-{
-    typedef char value_type;
+template <> struct cached_allocator<true> {
+  typedef char value_type;
 
-    thrust::device_vector<char> memory;
+  thrust::device_vector<char> memory;
 
-    char* allocate( std::ptrdiff_t num_bytes )
-    {
-        memory.resize( num_bytes );
-        return memory.data().get();
-    }
+  char *allocate(std::ptrdiff_t num_bytes) {
+    memory.resize(num_bytes);
+    return memory.data().get();
+  }
 
-    void deallocate( char* ptr, size_t )
-    {
-    }
+  void deallocate(char *ptr, size_t) {}
```

Comment:
Can you run clang format with the .clang-format file in the root of the project. This should make it easier to see what you have changed.